File: frontend/src/types/copilot.ts
import type { ChatHistoryRead } from "@/types/domain";

export interface CopilotConversationTurn {
  role: "user" | "assistant" | "system";
  content: string;
}

export interface CopilotCitation {
  document_name: string;
  section: string;
  source: string;
  page?: string | null;
  score: number;
  snippet: string;
  metadata: Record<string, unknown>;
}

export interface CopilotChatRequest {
  question: string;
  plant_id?: string | null;
  metadata_filters?: Record<string, string> | null;
  conversation_history?: CopilotConversationTurn[];
}

export interface CopilotChatResponse {
  summary: string;
  current_situation: string;
  evidence: string[];
  applicable_regulations: string[];
  recommendations: string[];
  citations: CopilotCitation[];
  confidence: number;
  provider: string;
  retrieved_documents: string[];
  saved_chat?: ChatHistoryRead | null;
}

export interface CopilotHistoryDeleteResponse {
  deleted_count: number;
}

File: backend/app/schemas/copilot.py
from typing import Any

from pydantic import BaseModel, Field

from app.schemas.domain import ChatHistoryRead


class CopilotConversationTurn(BaseModel):
    role: str = Field(pattern="^(user|assistant|system)$")
    content: str = Field(min_length=1)


class CopilotCitation(BaseModel):
    document_name: str
    section: str
    source: str
    page: str | None = None
    score: float
    snippet: str
    metadata: dict[str, Any] = Field(default_factory=dict)


class CopilotChatRequest(BaseModel):
    question: str = Field(min_length=3)
    plant_id: str | None = None
    user_id: str | None = None
    metadata_filters: dict[str, str] | None = None
    conversation_history: list[CopilotConversationTurn] = Field(default_factory=list)


class CopilotHistoryDeleteResponse(BaseModel):
    deleted_count: int


class CopilotChatResponse(BaseModel):
    summary: str
    current_situation: str
    evidence: list[str] = Field(default_factory=list)
    applicable_regulations: list[str]
    recommendations: list[str]
    citations: list[CopilotCitation]
    confidence: float
    provider: str
    retrieved_documents: list[str] = Field(default_factory=list)
    saved_chat: ChatHistoryRead | None = None
